pgx/src/datum/date.rs

```diff
@@ -43,6 +55,24 @@ impl Date {
     pub fn new(date: time::Date) -> Self {
         Date(date)
     }
+
+    pub fn infinity() -> Self {
+        Date(time::Date::MAX)
+    }
+
+    pub fn neg_infinity() -> Self {
+        Date(time::Date::MIN)
+    }
+
+    #[inline]
+    pub fn is_infinity(self) -> bool {
+        self.0 == time::Date::MAX
+    }
+
+    #[inline]
+    pub fn is_neg_infinity(self) -> bool {
+        self.0 == time::Date::MIN
+    }
```

Comment:
This encoding of infinity as two specific values of time::Date seems questionable to me. We allow DerefMut to the internal type. That may have been a bad decision, but that means that people can blithely assign a non-infinity value to the internal value and get infinity back. That's a correctness flaw.

It may be what Postgres does internally, reading their documentations, but I still don't think we should open the door to incorrect combinations of operations.

Reply:
I've made a separate PR for this with more explanation https://github.com/tcdi/pgx/pull/617   The Datums I use for +/- infinity are the correct PG values.
By substituting time::Date::MIN/time::Date::Max I figure anyone working with datum::Date would probably not encounter a breaking change infinite dates will always be greater non-infinite  (and vice versa for -infinity) 

Range<T> is dependent on this fix